core/editable.js

```diff
@@ -69,7 +84,7 @@
 					// Restore focus on element which we cached (on selectionCheck) as previously active.
 					active = this.editor._.previousActive || this.getDocument().getActive();
 					if ( this.contains( active ) ) {
-						active.focus();
+						active.focus( focusOptions );
```

Comment:
This invocation does not work as intended. `active` is an instance of `CKEDITOR.dom.element`, not an editable. In such case, the signature of the `focus` method is different. In fact passing this parameter causes `focus` to be called asynchronously, as is treated as `defer` attribute. Please see: https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_element.html#method-focus

That creates the tricky situation when the new fix is not working with nested editables (e.g. captions in `image2` or `easyimage` plugins), as viewport is always scrolled to these elements when inline editable is used (it's not present in classic, `iframe` based editor, as in such situation the `iframe` itself is a viewport, so only the content of `iframe` is scrolled, not the whole page).

I don't see any really sensible solution to that. We can overload `CKEDITOR.dom.element#focus` method to treat the fist parameter as `defer` one if value other than object is passed and as `focusOptions` if object is passed (in the second case we could make `defer` one of options, as probably there will be cases when both of the behaviours will be needed). It'd result in moving most of the logic from `CKEDITOR.editable` to `CKEDITOR.dom.element`. After that changes `CKEDITOR.editable#focus` will be using `CKEDITOR.dom.element#focus`, only adding some new behaviour.

However this change is not trivial anymore and requires much more tests. WDYT @engineering-this, @mlewand?